### PR TITLE
decode-string - Refael Gawi

### PR DESCRIPTION
**Pull request by refaelgawi**

### CODE REVIEW

**Comment by avishain** [comment]
CODE REVIEW
-----------------

The good:
- Modularity is great (the separation to few functions).
- Variable names are significant and valuable.
- Tests cover several scenarios.
- Use of es6.
- It seem to work as expected!!

Could be better:
- The  helper function is long and could have been split too.
- There are some unnecessary assigns (such as `const times: number = +timesStr;`, `const timesStrLength = timesStr.length;`, etc.)
- Could have reduce the amount of tests as there are many and some of them cover the same scenario

**Comment by Aymanw1998** [comment]
# CODE REVIEW
Could be better:
- There are a lot of tests, which can contain all the less testing options.
- You can not put all the files in here (just your_file.ts and your_file_test.ts)
- You have a lot of variables, there are some things that do not help mush, example: (times, timesStrLength), you could only use timesStr

Summary:
- Good job 
- Read and clean code
- Good luck

**Comment by sharon-cohen** [comment]
CODE REVIEW 

- The names of the variables and functions with meaning.

- Complex scenarios in the test

- Smart division auxiliary functions

For next time

- A clearer explanation for the task.
1. General presentation of the task with a simple example
2. Presenting a more complex example
3. Assumptions to be made for the task
4. Extreme scenarios

File: hw1_interview_questions/__tests__/yoav_zibin-test.ts [hunk 1]
```diff
@@ -1,17 +0,0 @@
-import {movingAvg} from '../yoav_zibin';
```
Comment (yoav-zibin) [inline hw1_interview_questions/__tests__/yoav_zibin-test.ts:1]: why deleted?

File: hw1_interview_questions/yoav_zibin.ts [hunk 2]
```diff
@@ -1,23 +0,0 @@
-/**
```
Comment (yoav-zibin) [inline hw1_interview_questions/yoav_zibin.ts:1]: same

**Comment by yoav-zibin** [comment]
thanks!

**Pull request closed without merging**